extensions/optaplanner/deployment/src/main/java/io/quarkus/optaplanner/deployment/OptaPlannerProcessor.java

```
@@ -0,0 +1,250 @@
+                    + " because the Quarkus extension scans too.\n"
+                    + "Maybe delete the scanAnnotatedClasses element in the solver config.");
+        }
+        if (solverConfig.getSolutionClass() == null) {
+            solverConfig.setSolutionClass(findSolutionClass(recorderContext, indexView));
+        }
+        if (solverConfig.getEntityClassList() == null) {
+            solverConfig.setEntityClassList(findEntityClassList(recorderContext, indexView));
+        }
+        if (solverConfig.getScoreDirectorFactoryConfig() == null) {
+            ScoreDirectorFactoryConfig scoreDirectorFactoryConfig = new ScoreDirectorFactoryConfig();
+            scoreDirectorFactoryConfig.setConstraintProviderClass(findConstraintProviderClass(recorderContext, indexView));
+            solverConfig.setScoreDirectorFactoryConfig(scoreDirectorFactoryConfig);
+        }
+        optaPlannerQuarkusConfig.solver.environmentMode.ifPresent(solverConfig::setEnvironmentMode);
+        optaPlannerQuarkusConfig.solver.moveThreadCount.ifPresent(solverConfig::setMoveThreadCount);
+        applyTerminationProperties(solverConfig);
+    }
+
+    private Class<?> findSolutionClass(RecorderContext recorderContext, IndexView indexView) {
+        Collection<AnnotationInstance> annotationInstances = indexView.getAnnotations(DotNames.PLANNING_SOLUTION);
+        if (annotationInstances.size() > 1) {
+            throw new IllegalStateException("Multiple classes (" + convertAnnotationInstancesToString(annotationInstances)
+                    + ") found with a @" + PlanningSolution.class.getSimpleName() + " annotation.");
+        }
+        if (annotationInstances.isEmpty()) {
+            throw new IllegalStateException("No classes (" + convertAnnotationInstancesToString(annotationInstances)
+                    + ") found with a @" + PlanningSolution.class.getSimpleName() + " annotation.");
```

We talked about removing this fail-fast, to follow hibernate-extension's approach, so I 'll do that in the morning - that is the only thing left. Just wanted to give CI a night to validate everything else is in place.

---

Open question, what is the behavior we want:
A) When there is no @PlanningSolution class, fail fast (current implementation, differs from hibernate extension)
B) When there is no @PlanningSolution class, only fail fast if there are any `quarkus.optaplanner.*` application.properties, otherwise do nothing
C) When there is no @PlanningSolution class, even if there are any `quarkus.optaplanner.*` application.properties, always do nothing quietly

---

I presume A) is a PITA when doing demo's where you grab something from code.quarkus.io and it doesn't just run?